### PR TITLE
[Relay] Add dynamic SparseToDense

### PR DESCRIPTION
**Pull request by lixiaoquan**
@mbrookhart @electriclilies Please help to review, thanks.

### CODE REVIEW

**Review by electriclilies — COMMENTED** [review]
Overall looks good to me, you just need to change some ICHECKs to CHECKs in your type rel

**Review by mbrookhart — APPROVED** [review]

File: src/relay/op/dyn/tensor/transform.cc [hunk 1]
```diff
@@ -556,6 +556,77 @@ Examples::
...
+  auto output_shape = types[3].as<TensorTypeNode>();
+
+  if (sparse_indices == nullptr || sparse_values == nullptr || default_value == nullptr ||
+      output_shape == nullptr) {
+    return false;
+  }
+
+  ICHECK(sparse_indices->dtype.is_int()) << "sparse_indices must be tensor of integers";
+
+  ICHECK_LE(sparse_indices->shape.size(), 3)
```
[the remark this answers is not in the document]
Comment (lixiaoquan) [reply src/relay/op/dyn/tensor/transform.cc]: Fixed

**Review by electriclilies — APPROVED** [review]
LGTM, thanks!

**Comment by lixiaoquan** [comment]
@zhiics Could you please help merge this? Thanks

**Review by zhiics — APPROVED** [review]

**Comment by zhiics** [comment]
Thanks @lixiaoquan @mbrookhart @electriclilies

**Pull request merged**